README.md

```diff
@@ -15,47 +16,115 @@
+This API would be a new identity scheme within [IdentityCredential](https://github.com/agl/identity-credential) using the scheme identifier `mdoc`.
 
+### Requests
 
-### Request
+```webidl
+partial dictionary IdentityProviderConfig  {
+  optional MdocIdentityProviderConfig mdoc;
+}
 
-The website would need to provide a document type and desired data elements, as well as a requester identity object that allows the device to extract the encryption public key (and the mdoc to optionally verify the eligibility of the website to request the mdoc presentment).
+dictionary MdocIdentityProviderConfig {
+  required DOMString nonce;
+
+  // The mdoc document type requested. See ISO 18013-5 section 7.1.
+  required DOMString documentType;
+
+  // The elements of the mdoc requested.
+  required sequence<MdocElement> requestedElements;
+
+  // The number of days that the requester will keep the data for.
+  // Not providing this is equivalent to 0, meaning not asking to store.
+  // Infinity means forever.
+  double retentionDays;
+
+  // A set of reader identities. Each value is a base64-encoded sequence of one
+  // or more X.509 certificates in ASN.1 DER encoding (a “chain” of
+  // certificates). At least one value must be provided. The first element of
+  // each X.509 chain MUST be the end-entity certificate. Subsequent X.509
+  // certificates in a chain are intermediate certificates as may help the wallet
+  // verify the end-entity certificate.
+  //
+  // If the requester has no applicable X.509 chain then it may use a
+  // single, self-signed certificate.
+  //
+  // The underlying wallet will evaluate these certificate chains
+  // and pick one, at its discretion, to identify the requesting entity. The
+  // SubjectPublicKeyInfo of the end-entity certificate of the selected chain
+  // is used to encrypt the response.
+  required sequence<DOMString> readerAuthentication;
+};
+
+dictionary MdocElement {
+  required DOMString namespace;  // As defined in ISO 18013-5 section 7.
+  required DOMString name;
+};
+```
+
+An example request:
+
+```js
+let request = {
+  identity: {
+    providers: [{
+      mdoc: {
+        nonce: "gf69kepV+m5tGxMyMF0cnn9NCnRHez/LUIsFtLi6pwg=",
+        documentType: "org.iso.18013.5.1.mDL",
+        retentionDays: 90,
+        readerPublicKey: ["ftl+VEHPB17r2oi6it3ENaqhOOB0AZbAkb5f4VlCPakpdNioc9QZ7X/6w..."],
```

Comment:
This looks base64-encoded, but do we know any more than that? Is it a SEC 1 uncompressed EC key? Something else?